homeassistant/components/casatunes/media_player.py

```diff
@@ -0,0 +1,270 @@
+"""
+Support for interfacing with CasaTunes devices.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/media_player.casatunes/
+"""
+import logging
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.media_player import (
+    MediaPlayerDevice, PLATFORM_SCHEMA)
+from homeassistant.components.media_player.const import (
+    MEDIA_TYPE_MUSIC,
+    SUPPORT_SELECT_SOURCE, SUPPORT_TURN_ON, SUPPORT_TURN_OFF,
+    SUPPORT_VOLUME_SET, SUPPORT_VOLUME_MUTE, SUPPORT_NEXT_TRACK,
+    SUPPORT_PREVIOUS_TRACK, SUPPORT_SEEK, SUPPORT_PLAY, SUPPORT_PAUSE,
+    SUPPORT_STOP)
+from homeassistant.const import (
+    CONF_HOST, CONF_PORT, CONF_NAME, STATE_IDLE, STATE_OFF, STATE_ON,
+    STATE_PAUSED, STATE_PLAYING)
+
+REQUIREMENTS = ['casatunes==0.0.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'CasaTunes'
+DEFAULT_PORT = 8735
+DEFAULT_TIMEOUT = 10
+
+SUPPORT_CT_ZONE = SUPPORT_SELECT_SOURCE | SUPPORT_TURN_OFF | \
+                  SUPPORT_TURN_ON | SUPPORT_VOLUME_SET | SUPPORT_VOLUME_MUTE
+
+SUPPORT_CT_PLAYER = SUPPORT_NEXT_TRACK | SUPPORT_PREVIOUS_TRACK | \
+                    SUPPORT_SEEK | SUPPORT_PLAY | SUPPORT_PAUSE | SUPPORT_STOP
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+})
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up the CasaTunes platform."""
+    from casatunes import CasaTunes
+
+    host = config.get(CONF_HOST)
+    port = config.get(CONF_PORT)
+
+    ct = CasaTunes(host, port)
+
+    sources = await ct.get_sources()
+    zones = await ct.get_zones()
+
+    zone_devices = []
+    player_devices = []
+
+    if sources is False:
+        _LOGGER.error("Sources response is incorrect")
+        return False
+
+    if zones is False:
+        _LOGGER.error("Zones response is incorrect")
+        return False
+
+    for zone in zones:
+        try:
+            zone_devices.append(CasaTunesZone(zone))
+        except TypeError:
```

Comment:
Under what conditions would this error be thrown?  Do we want the platform to retry if the initial attempt doesn't work?

Reply:
There could (hypothetically) be a network issue or CasaTunes could be updating.

Reply:
In that case you may want to throw a PlatformNotReady error that way the Home Assistant system can retry in a few seconds to try and setup the platform.  Without that, a user will need to restart home assistant to try getting this platform to work.

Reply:
Let's wait for Travis to run his tests again